### ISSUE TITLE
literal IPv6 addresses are not parsed correctly in lb_endpoints with STRICT_DNS

### ISSUE THREAD

**Issue by geokb** [envoyproxy/envoy#18606]
I want to have cluster lb_endpoints that may include both hostnames and literal IP addresses at the same time. When the cluster's type is `STRICT_DNS`, the configuration is refused with this error keeps iterating in the logs
```
[2021-10-13 13:53:47.264][1][warning][config] [source/common/config/grpc_subscription_impl.cc:126] gRPC config for type.googleapis.com/envoy.config.cluster.v3.Cluster rejected: Error adding/updating cluster(s) default-c02-tcp-443-cluster: malformed url: tcp://fdee:6207::1000:0:0:a3bd:22607
```

I guess this is because the square brackets are not set around the IPv6 literal (i.e. it should be `tcp://[fdee:6207::1000:0:0:a3bd]:22607`)? I tried having IPv4 literals and they work perfectly fine. So this is an IPv6-specific issue.

Also I tried it with `LOGICAL_DNS` but I got the error
```
[2021-10-13 13:33:38.163][1][warning][config] [source/common/config/grpc_subscription_impl.cc:126] gRPC config for type.googleapis.com/envoy.config.cluster.v3.Cluster rejected: Error adding/updating cluster(s) default-c02-tcp-443-cluster: LOGICAL_DNS clusters must have a single locality_lb_endpoint and a single lb_endpoint
```
But that was expected since I have more than one endpoint.

Also I am using Envoy 1.19 if that may help.

**Comment by geokb** [comment]
@alyssawilk thanks for the response. What I meant is that it seems Envoy always blindly assumes that the `address` field of the `core.SocketAddress` message is a hostname in the case of the `STRICT_DNS` and does not add the brackets if the field contains an IPv6 literal. That's why it seems it's not a problem if the `address` is an IPv4 address.

In my case, I am having more than one lb_endpoint. Some of those endpoints can be IPv6 as well as IPv4 literals along side with the resolvable hostnames.

The weird thing is that if the lb_endpoints only include one or more IPv6 literals, this issue disappears. In other words, this bug only appears if there is a hostname in the lb_endpoints.

**Comment by tpetkov-VMW** [comment]
I'm willing to work on this, if it's still considered a bug.

**Comment by dimitraPat** [comment]
Hi,
Is there any update on this issue?
Thanks

**Comment by jewertow** [comment]
This issue was fixed in v1.23. @alyssawilk could you close it?
